### ISSUE TITLE
[$1000] Account name not updating after renaming

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#19991]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open new two accounts
2. Send a money request from account A to account B
3. Observe account B on LHN which is an email
4. Go to account B and rename it to anything
5. Open account A and observe the name of account B on LHN
## Expected Result: 
Name should update on LHN after renaming account B
## Actual Result: 
Name does not update on LHN after renaming
NOTE : This bug occurs if only account A sends money request to account B before starting any report or posting any report action.

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.22-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/cf22976c-e99d-438a-8faf-05c31dcbcffa

https://github.com/Expensify/App/assets/43996225/0da4a490-4d10-4aef-be79-1cb8030b6175

**Expensify/Expensify Issue URL:**
**Issue reported by:** @nathan-mulugeta
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1685439541539559

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01c1c4bd705231f1f8</li>
        <li>Upwork Job ID: 1664664071789846528</li>
        <li>Last Price Increase: 2023-06-02</li>
    </ul>
</details>

**Comment by s-alves10** [comment]
This looks like the issue in backend

**Comment by flaviadefaria** [comment]
@s-alves10 can you clarify what you mean by that? Is there another issue related to this?

**Comment by s-alves10** [comment]
> @s-alves10 can you clarify what you mean by that? Is there another issue related to this?

This issue is related to syncing by push events from server.
By my observation in the above scenario, A doesn't receive any events when B renames.
This is the issue of backend, I think.

**Comment by Julesssss** [comment]
This is valid, but I don't think it's something we care enough about to fix. 

The next time the user opens the app, the names should be updated. So it's not necessary to update the name immediately.
